account_payment_order/tests/test_payment_order_inbound.py

```diff
@@ -136,3 +137,23 @@ def test_creation(self):
         payment_order.cancel2draft()
         payment_order.unlink()
         self.assertEqual(len(self.payment_order_obj.search(self.domain)), 0)
+
+    def test_payment_partner_bank_value(self):
+        payment_order = self.inbound_order
+        self.assertEqual(len(payment_order.ids), 1)
+        payment_order.write({"journal_id": self.journal.id})
+        self.assertEqual(len(payment_order.payment_line_ids), 1)
+        self.assertFalse(payment_order.payment_ids)
+        # Open payment order
+        payment_order.draft2open()
+        self.assertEqual(len(payment_order.payment_ids), 1)
+        payment = payment_order.payment_ids[0]
+        self.assertEqual(len(payment.payment_line_ids), 1)
+        payment_line = payment.payment_line_ids[0]
+        self.assertEqual(payment.partner_bank_id, payment_line.partner_bank_id)
+        # Force payment's available banks compute to check parner_bank_id
+        # fields value doesn't change
+        with Form(payment) as payment_form:
+            payment_form.partner_id = self.env["res.partner"]
+            payment_form.partner_id = payment_line.partner_id
```

Comment:
This test case is still a non real one and doesn't reflect a business case. What are you trying to accomplish? Maybe what we have to do is to forbid changing account.payment linked to payment orders.

Reply:
Modified tests to a more "likely" use case un which a user can accidentaly modify a field on `account.payment` model's form view. `partner_bank_id` field would get an incorrect value.

I recorded a video reproducing the issue. In this case field turns to False because journal has no bank: 

https://github.com/OCA/bank-payment/assets/31987854/894d3bbb-da7e-4c0d-a38e-5fabd3f417c3

Reply:
Hi @pedrobaeza is it ok?

Reply:
Thanks for the video. I think the proper fix should be to block any change (through an override in the `write` method of the `account.payment`) if it's linked to a payment order.